docs/tutorials/cooperative_groups_tutorial.rst

```diff
@@ -0,0 +1,312 @@
+
+This tutorial will show you the basic concepts of the cooperative groups in HIP
+programming model, the most essential tooling around it and briefly rehash some
+commonalities of heterogenous APIs in general. Mild familiarity with the C/C++
+compilation model and the language is assumed throughout this article.
+
+Prerequisites
+=============
+
+In order to follow this tutorial you will need properly installed drivers and a
+HIP compiler toolchain to compile your code. Because HIP provided by ROCm
+supports compiling and running on Linux and Windows with AMD and NVIDIA GPUs
+alike, the combination of install instructions are more then worth covering as
+part of this tutorial. Please refer to :doc:`/install/install` on how to
+install HIP development packages.
+
+Simple HIP Code
+===============
+
+.. TODO: Add link here to SAXPY and subsections
+
+To get familiar with the Heterogenous programming, you should check the SAXPY
+tutorial and the first HIP code subsection. The compiling is also well described
+in that tutorial.
+
+Tiled partition
+===============
+
+The tiled partition can be used to calculate the sum of ``partition_size``
+length sequences and sum of ``result_size``/ ``BlockSize`` length sequences. The
+host side reference implementation is the following:
+
+.. code-block:: cpp
+
+  // Host side function to perform the same reductions as executed on the GPU
+  std::vector<unsigned int> ref_reduced(const unsigned int        partition_size,
+                                        std::vector<unsigned int> input)
+  {
+      const unsigned int        input_size  = input.size();
+      const unsigned int        result_size = input_size / partition_size;
+      std::vector<unsigned int> result(result_size);
+
+      for(unsigned int i = 0; i < result_size; i++)
+      {
+          unsigned int partition_result = 0;
+          for(unsigned int j = 0; j < partition_size; j++)
+          {
+              partition_result += input[partition_size * i + j];
+          }
+          result[i] = partition_result;
+      }
+
+      return result;
+  }
+
+Device side code
+----------------
+
+.. TODO: Add link here to reduction tutorial and subsections
+
+To be able to calculate the sum of the sets of numbers, the tutorial using the
+shared memory based reduction at device side. In this tutorial the warp level
+intrinsics usage not covered like in the reduction tutorial. The ``x`` input
+variable is a shared pointer, which needs to be synchronized after every value
+changes. The ``thread_group`` input parameter can be ``thread_block_tile`` or
+``thread_block`` also, because the ``thread_group`` is the parent class of these
+types. The ``val`` is the numbers, which we wants to calculate the sum of. The
+returned results of this function return the final results of the reduction on
+thread ID 0 of the ``thread_group`` and at every other threads the function
+results are 0.
```

Comment:
```suggestion
To calculate the sum of the sets of numbers, the tutorial uses the shared memory-based reduction on the device side. The warp level intrinsics usage is not covered in this tutorial, unlike in the Reduction tutorial. The x input variable is a shared pointer, which needs to be synchronized after every value changes. The ``thread_group`` input parameter can be ``thread_block_tile`` or
``thread_block`` because the ``thread_group`` is the parent class of these
types. The ``val`` are the numbers to calculate the sum of. The
returned results of this function return the final results of the reduction on
thread ID 0 of the ``thread_group``, and at every other thread, the function
results are 0.
```